assets/js/base/components/quantity-selector/index.js

```diff
@@ -116,11 +120,11 @@ const QuantitySelector = ( {
 					'Increase quantity',
 					'woo-gutenberg-products-block'
 				) }
-				disabled={ disabled }
+				disabled={ disabled || ! canIncrease }
 				className="wc-block-quantity-selector__button wc-block-quantity-selector__button--plus"
 				onClick={ () => {
-					const newQuantity = currentValue + 1;
-					setCurrentValue( newQuantity );
+					const newQuantity = quantity + 1;
+					onChange( newQuantity );
```

Comment:
I know this already existed, but I don't see the need for defining a new constant here, you can just provide `quantity +1` as an argument directly.
```suggestion
					onChange( quantity + 1 );
```

Reply:
`newQuantity` is used twice in each `onClick` handler - `onChange` to set the value and then `speak` for accessibility, e.g. "Quantity increased to 3.". So leaving this as is :)